### ISSUE TITLE
"Error: 409-BucketNotEmpty is not empty" when updating ELZ deployment

### ISSUE THREAD

**Issue by agtpgh** [oracle-quickstart/oci-landing-zones#44]
Hi,

We have deployed the ELZ for testing. When attempting to make a change to the ELZ (in this case updating VCN CIDR blocks) we receive error 409 for various log buckets.

It seems when we run a 'terraform apply' it tries to delete these as we see this terminal output:

```
module.prod_environment.module.logging.module.service_event_log_bucket.oci_objectstorage_bucket.bucket: Destroying...
module.nonprod_environment.module.logging.module.default_log_bucket.oci_objectstorage_bucket.bucket: Destroying...
module.nonprod_environment.module.logging.module.service_event_log_bucket.oci_objectstorage_bucket.bucket: Destroying...
module.nonprod_environment.module.logging.module.audit_log_bucket.oci_objectstorage_bucket.bucket: Destroying...
module.prod_environment.module.logging.module.audit_log_bucket.oci_objectstorage_bucket.bucket: Destroying...
```

We are not attempting to change any logging or bucket related settings. Our change was to update 'terraform.tfvars' network sections, for example:

```
prod_hub_vcn_cidr_block            = "10.100.0.0/16"
prod_hub_public_subnet_cidr_block  = "10.100.1.0/24"
prod_hub_private_subnet_cidr_block = "10.100.2.0/24"
```

Is updated to become:

```
prod_hub_vcn_cidr_block            = "10.150.0.0/16"
prod_hub_public_subnet_cidr_block  = "10.150.1.0/24"
prod_hub_private_subnet_cidr_block = "10.150.2.0/24"
```

When we run the apply, we receive the errors as indicated, full output shown below (minus OPC).

```
Error: 409-BucketNotEmpty, Bucket named 'ELZ_N_auditLogs_standard' is not empty. Delete all object versions first.
│ Suggestion: The resource is in a conflicted state. Please retry again or contact support for help with service: Object Storage Bucket
│ Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/object_storage_bucket
│ API Reference: https://docs.oracle.com/iaas/api/#/en/objectstorage/20160918/Bucket/DeleteBucket
│ Request Target: DELETE https://objectstorage.uk-london-1.oraclecloud.com/n/lr8pvvcfqwfw/b/ELZ_N_auditLogs_standard
│ Provider version: 4.96.0, released on 2022-10-05. This provider is 34 Update(s) behind to current.
│ Service: Object Storage Bucket
│ Operation Name: DeleteBucket
│ OPC request ID: 
│
│
│
╵
╷
│ Error: 409-BucketNotEmpty, Bucket named 'ELZ_P_defaultLogs_standard' is not empty. Delete all object versions first.
│ Suggestion: The resource is in a conflicted state. Please retry again or contact support for help with service: Object Storage Bucket
│ Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/object_storage_bucket
│ API Reference: https://docs.oracle.com/iaas/api/#/en/objectstorage/20160918/Bucket/DeleteBucket
│ Request Target: DELETE https://objectstorage.uk-london-1.oraclecloud.com/n/lr8pvvcfqwfw/b/ELZ_P_defaultLogs_standard
│ Provider version: 4.96.0, released on 2022-10-05. This provider is 34 Update(s) behind to current.
│ Service: Object Storage Bucket
│ Operation Name: DeleteBucket
│ OPC request ID: 
│
│
│
╵
╷
│ Error: 409-BucketNotEmpty, Bucket named 'ELZ_N_defaultLogs_standard' is not empty. Delete all object versions first.
│ Suggestion: The resource is in a conflicted state. Please retry again or contact support for help with service: Object Storage Bucket
│ Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/object_storage_bucket
│ API Reference: https://docs.oracle.com/iaas/api/#/en/objectstorage/20160918/Bucket/DeleteBucket
│ Request Target: DELETE https://objectstorage.uk-london-1.oraclecloud.com/n/lr8pvvcfqwfw/b/ELZ_N_defaultLogs_standard
│ Provider version: 4.96.0, released on 2022-10-05. This provider is 34 Update(s) behind to current.
│ Service: Object Storage Bucket
│ Operation Name: DeleteBucket
│ OPC request ID: 
│
│
│
╵
╷
│ Error: 409-BucketNotEmpty, Bucket named 'ELZ_P_serviceEvents_standard' is not empty. Delete all object versions first.
│ Suggestion: The resource is in a conflicted state. Please retry again or contact support for help with service: Object Storage Bucket
│ Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/object_storage_bucket
│ API Reference: https://docs.oracle.com/iaas/api/#/en/objectstorage/20160918/Bucket/DeleteBucket
│ Request Target: DELETE https://objectstorage.uk-london-1.oraclecloud.com/n/lr8pvvcfqwfw/b/ELZ_P_serviceEvents_standard
│ Provider version: 4.96.0, released on 2022-10-05. This provider is 34 Update(s) behind to current.
│ Service: Object Storage Bucket
│ Operation Name: DeleteBucket
│ OPC request ID: 
│
│
│
╵
╷
│ Error: 409-BucketNotEmpty, Bucket named 'ELZ_P_auditLogs_standard' is not empty. Delete all object versions first.
│ Suggestion: The resource is in a conflicted state. Please retry again or contact support for help with service: Object Storage Bucket
│ Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/object_storage_bucket
│ API Reference: https://docs.oracle.com/iaas/api/#/en/objectstorage/20160918/Bucket/DeleteBucket
│ Request Target: DELETE https://objectstorage.uk-london-1.oraclecloud.com/n/lr8pvvcfqwfw/b/ELZ_P_auditLogs_standard
│ Provider version: 4.96.0, released on 2022-10-05. This provider is 34 Update(s) behind to current.
│ Service: Object Storage Bucket
│ Operation Name: DeleteBucket
│ OPC request ID: 
│
│
│
╵
╷
│ Error: 409-BucketNotEmpty, Bucket named 'ELZ_N_serviceEvents_standard' is not empty. Delete all object versions first.
│ Suggestion: The resource is in a conflicted state. Please retry again or contact support for help with service: Object Storage Bucket
│ Documentation: https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/object_storage_bucket
│ API Reference: https://docs.oracle.com/iaas/api/#/en/objectstorage/20160918/Bucket/DeleteBucket
│ Request Target: DELETE https://objectstorage.uk-london-1.oraclecloud.com/n/lr8pvvcfqwfw/b/ELZ_N_serviceEvents_standard
│ Provider version: 4.96.0, released on 2022-10-05. This provider is 34 Update(s) behind to current.
│ Service: Object Storage Bucket
│ Operation Name: DeleteBucket
│ OPC request ID: 
```

**Comment by agtpgh** [comment]
It looks like this might be the same/similar issue as shown in #40 that you are tracking as LANZ-1758.

**Comment by paulmuah** [comment]
Greetings @agtpgh. We are researching with the TF Provider to see if there's a permanent fix for this issue. I will keep you in the loop once we get a respond and a fix.

**Comment by paulmuah** [comment]
Issue resolved. Please refer to Closed Issue #40

**Comment by pathaknikhil87** [comment]
[VinayKumar611](https://github.com/VinayKumar611)  Even with the new code, I am facing the same issue. I am using the latest available "oci-landing-zones" code..
